src/native/windows/apps/prunsrv/prunsrv.c

```diff
@@ -952,9 +955,16 @@ static BOOL docmdStopService(LPAPXCMDLINE lpCmdline)
         if (!rv) {
             /* Wait for the timeout if any */
             int  timeout     = SO_STOPTIMEOUT;
+            if (timeout == 0) {
+                /* waiting for ever doesn't look OK here */
+                timeout = ONE_MINUTE_SEC;
+            }
             if (timeout) {
+                /* the SO_STOPTIMEOUT applies to the stop command and to the time service needs to stop */
+                /* We also add TIMEFORSERVICEMANAGER seconds for the service logic */
                 int i;
-                for (i = 0; i < timeout; i++) {
+                for (i = 0; i < (timeout*2+TIMEFORSERVICEMANAGER); i++) {
```

Comment:
This makes the minimum timeout 17s. That seems high to me.

I would expect to set a timeout for the total time (sending the stop command, waiting for the service to stop, overhead). How practical is to set set a `maximumFinishTime` here of `currentTime + timeout` and then calculate the `timeout`  as `maximumFinishTime - currentTime` whenever a timeout is required?

Reply:
Agreed that makes more sense.
Here we just need to wait for timeout, the service is another process where we need maximumFinishTime and currentTime.